### PR TITLE
feat: use customization to deduce typings

### PR DESCRIPTION
**Pull request by realSpok**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/cloud-toolkit/src/services/update-typings.ts [hunk 1]
```diff
@@ -21,5 +25,30 @@ export default async function updateTypings(
...
+  const builtCustomizationPath = `${path.resolve(rootPath)}/dist/code-customizations/index.js`;
+
+  try {
+    await fs.access(builtCustomizationPath);
+    // eslint-disable-next-line max-len
+    // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
+    const customizer = require(builtCustomizationPath).default;
+
+    try {
+      customizer(agent);
```
[the remark this answers is not in the document]
Comment (Thenkei) [reply packages/cloud-toolkit/src/services/update-typings.ts:38]: 💪

[the remark this answers is not in the document]
Comment (Thenkei) [reply packages/cloud-toolkit/src/services/update-typings.ts:38]: I said bullshits. We don't need to compute **no code customizations** they are not accessible to the client. 🙏

File: packages/cloud-toolkit/src/services/update-typings.ts [hunk 2]
```diff
@@ -21,5 +25,30 @@ export default async function updateTypings(
       introspection: await httpForestServer.getIntrospection(),
     }),
   );
+
+  const builtCustomizationPath = `${path.resolve(rootPath)}/dist/code-customizations/index.js`;
+
+  try {
+    await fs.access(builtCustomizationPath);
```
[the remark this answers is not in the document]
Comment (realSpok) [reply packages/cloud-toolkit/src/services/update-typings.ts:32]: I'm actually using `fs/promises` library which doesn't expose `existsSync`

[the remark this answers is not in the document]
Comment (Thenkei) [reply packages/cloud-toolkit/src/services/update-typings.ts:32]: `fs` expose it. And it's native too. :)

File: packages/cloud-toolkit/src/services/update-typings.ts [hunk 3]
```diff
@@ -21,5 +25,30 @@ export default async function updateTypings(
       introspection: await httpForestServer.getIntrospection(),
     }),
   );
+
```
Comment (Thenkei) [inline packages/cloud-toolkit/src/services/update-typings.ts:28]:
```suggestion
export default async function loadCustomization(agent: Agent, path: string) {
  if (!fs.existsSync(path)) {
    logger('Info', `Customization file not found at ${path}`);

    return;
  }

  const customization = require(path);

  const entryPoint = customization?.default || customization;

  if (typeof entryPoint !== 'function') {
    throw new Error('Customization file must export a function');
  }

  await entryPoint(agent);
}
```
https://github.com/ForestAdmin/cloud-agent-manager/blob/0963fe110efdfbf4d9835d5652e554d59295cb23/packages/lambda-layers/src/agent/internal/load-customization.ts#L9 

This is how we handle it in the lambda, the code should be pretty similar.

**Pull request merged**